src/main/java/com/amihaiemil/versioneye/RtRepository.java

```diff
@@ -50,4 +54,63 @@
     RtRepository(final JsonObject repository) {
         this.repository = repository;
     }
+
+    @Override
+    public String name() {
+        return this.repository.getString("name");
+    }
+    
+    @Override
+    public String fullname() {
+        return this.repository.getString("fullname");
+    }
+
+    @Override
+    public String language() {
+        return this.repository.getString("language");
+    }
+
+    @Override
+    public String ownerLogin() {
+        return this.repository.getString("owner_login");
+    }
+
+    @Override
+    public String ownerType() {
+        return this.repository.getString("owner_type");
+    }
+
+    @Override
+    public String description() {
+        return this.repository.getString("description");
+    }
+
+    @Override
+    public boolean isPrivate() {
+        return this.repository.getBoolean("private");
+    }
+
+    @Override
+    public boolean fork() {
+        return this.repository.getBoolean("fork");
+    }
+
+    @Override
+    public List<String> branches() {
+        List<String> branches = new ArrayList<String>();
+        try {
+            final JsonArray jsonBranches =
+                this.repository.getJsonArray("branches");
+            for(int i = 0; i < jsonBranches.size(); i++) {
+                branches.add(jsonBranches.getString(i));
+            }
+        } catch(final ClassCastException castException) {
```

Comment:
@SherifWaly Are you sure there isn't any other way? If not, I'm thinking if we shouldn't rethrow this exception wrapped inside an ``IllegalStateException`` -- because any Github repo must have at least the branch master, right? Otherwise it practically does not exist :) What do you think?